eval/lib/jwt.go

```diff
@@ -13,30 +13,123 @@ import (
 	"github.com/zclconf/go-cty/cty/function"
 	"github.com/zclconf/go-cty/cty/function/stdlib"
 
+	acjwt "github.com/avenga/couper/accesscontrol/jwt"
 	"github.com/avenga/couper/config"
+	"github.com/avenga/couper/config/reader"
 	"github.com/avenga/couper/internal/seetie"
 )
 
 const FnJWTSign = "jwt_sign"
 
 var rsaParseError = &rsa.PrivateKey{}
 
-func NewJwtSignFunction(jwtSigningProfiles []*config.JWTSigningProfile, confCtx *hcl.EvalContext) function.Function {
-	signingProfiles := make(map[string]*config.JWTSigningProfile)
-	rsaKeys := make(map[string]*rsa.PrivateKey)
+type JWTSigningConfig struct {
+	Claims             config.Claims
+	Key                interface{}
+	Name               string
+	SignatureAlgorithm string
+	TTL                time.Duration
+}
 
-	for _, sp := range jwtSigningProfiles {
-		signingProfiles[sp.Name] = sp
-		if strings.HasPrefix(sp.SignatureAlgorithm, "RS") {
-			key, err := jwt.ParseRSAPrivateKeyFromPEM(sp.KeyBytes)
-			if err != nil {
-				rsaKeys[sp.Name] = rsaParseError
-				continue
-			}
-			rsaKeys[sp.Name] = key
+func checkData(ttl, signatureAlgorithm string) (time.Duration, acjwt.Algorithm, error) {
+	var (
+		dur      time.Duration
+		parseErr error
+	)
+
+	alg := acjwt.NewAlgorithm(signatureAlgorithm)
+	if alg == acjwt.AlgorithmUnknown {
+		return dur, alg, fmt.Errorf("algorithm is not supported")
+	}
+
+	if ttl != "0" {
```

Comment:
ttl is of type `duration`. `0` is an invalid value. `0s`, `0ms`, ... is valid.

Reply:
This was implemented a long time ago in order to facilitate non-expiring tokens. That's why it was not documented as `duration` but as `string`.